plugin/federation/federation.go

```diff
@@ -0,0 +1,335 @@
+package federation
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/99designs/gqlgen/codegen"
+	"github.com/99designs/gqlgen/codegen/config"
+	"github.com/99designs/gqlgen/codegen/templates"
+	"github.com/99designs/gqlgen/plugin"
+	"github.com/vektah/gqlparser"
+	"github.com/vektah/gqlparser/ast"
+	"github.com/vektah/gqlparser/formatter"
+)
+
+type federation struct {
+	SDL      string
+	Entities []*Entity
+}
+
+// New returns a federation plugin that injects
+// federated directives and types into the schema
+func New() plugin.Plugin {
+	return &federation{}
+}
+
+func (f *federation) Name() string {
+	return "federation"
+}
+
+func (f *federation) MutateConfig(cfg *config.Config) error {
+	entityFields := map[string]config.TypeMapField{}
+	for _, e := range f.Entities {
+		entityFields[e.ResolverName] = config.TypeMapField{Resolver: true}
+		for _, r := range e.Requires {
+			if cfg.Models[e.Name].Fields == nil {
+				model := cfg.Models[e.Name]
+				model.Fields = map[string]config.TypeMapField{}
+				cfg.Models[e.Name] = model
+			}
+			cfg.Models[e.Name].Fields[r.Name] = config.TypeMapField{Resolver: true}
+		}
+	}
+	builtins := config.TypeMap{
+		"_Service": {
+			Model: config.StringList{
+				"github.com/99designs/gqlgen/graphql/introspection.Service",
```

Comment:
Can we move `github.com/99designs/gqlgen/graphql/introspection.Service` to `github.com/99designs/gqlgen/plugin/federation/federation.Service` ?